File: bytedance/dy/190.java
// 190. 颠倒二进制位

public class Solution {
    // 从低到高，按位颠倒
    public int reverseBits(int n) {
        int ans = 0;
        for (int i = 0; i < 32; i++) {
            int res = (n >> i) & 1; // 同0异1，适用于颠倒
            if (res == 1) {
                ans |= (1 << (31 - i)); // 加到结果上
            } 
        }
    }
}

File: bytedance/190.py
# 190. 颠倒二进制位
# 位运算的基本问题

# 取出 n 的最低位，加入结果 res 中。然后 n 右移，res 左移。循环此过程
class Solution:
    def reverseBits(self, n: int) -> int:
        res, count = 0, 32
        while count:
            res <<= 1
            # 取出 n 的最低位数加到 res 中
            res += n&1
            n >> 1
            count -= 1

        # return int(bin(res), 2)
        return res

x = Solution()
print(x.reverseBits(43261596))